ql/instruments/swaption.hpp

```diff
@@ -41,6 +41,11 @@ namespace QuantLib {
         enum Type { Physical, Cash };
     };
 
+    //! %settlement method
```

Comment:
If it weren't for the conflicting `Physical` labels, this could be a `Method` enumeration inside `Settlement`.  Any chance you can find a variation of the name?

Reply:
No problem, we could just have Physical, Cash_ParYieldCurve and Cash_CollateralizedCashPrice as enum values in Settlement::Type, is that what you mean? But that wouldn't be backward compatible, is that an issue?

Reply:
ah

Reply:
well, the ISDA definitions would suggest Physical_OTC and Physical_Cleared, but I don't see that makes sense for us, since we would treat them identically in the pricing

Reply:
We could add Default. For Cash Settlement we could use this as a replacement for boost::none as well. Would that make sense?

Reply:
We can call it future-proof. 😄 

I'd go for `PhysicalOTC` and `PhysicalCleared`, though (for consistency with the other two).

Reply:
Re `Default`: possibly.

Reply:
let's do PhysicalOTC and PhysicalCleared then, I'll amend the PR